R/pkg/R/serialize.R

```diff
@@ -58,7 +58,12 @@ writeObject <- function(con, object, writeType = TRUE) {
   # Checking types is needed here, since 'is.na' only handles atomic vectors,
   # lists and pairlists
   if (type %in% c("integer", "character", "logical", "double", "numeric")) {
-    if (is.na(object)) {
+    if (is.na(object[[1]])) {
+      # Uses the first element for now to keep the behavior same as R before
+      # 4.2.0. This is wrong because we should differenciate c(NA) from a
+      # single NA as the former means array(null) and the latter means null
+      # in Spark SQL. However, it requires non-trivial comparison to distinguish
```

Comment:
e.g.) we should check if the input is vector, list, array, etc, which is exactly being done at `getSerdeType`. However, this comparison here (up to my best knowledge) is a shortcut to avoid the overhead from `getSerdeType`. So, I just decided to leave it as is for now.